libsolidity/Parser.cpp

```diff
@@ -263,6 +266,39 @@ ASTPointer<StructDefinition> Parser::parseStructDefinition()
 	return nodeFactory.createNode<StructDefinition>(name, members);
 }
 
+ASTPointer<EnumValue> Parser::parseEnumDeclaration()
+{
+	ASTNodeFactory nodeFactory(*this);
+	ASTPointer<ASTString> name = expectIdentifierToken();
+	nodeFactory.markEndPosition();
```

Comment:
Don't worry I have my fair share of crappy mechanisms under my belt too. We can try to see if we can make it a bit more intuitive in a small PR.